File: RouterLib/src/main/java/com/kronos/router/interceptor/CacheInterceptor.java
package com.kronos.router.interceptor;

import android.util.Log;

import com.kronos.router.exception.RouteNotFoundException;
import com.kronos.router.model.RouterParams;

import java.util.Map;

public class CacheInterceptor implements Interceptor {
    private final Map<String, RouterParams> cachedRoutes;

    CacheInterceptor(Map<String, RouterParams> cachedRoutes) {
        this.cachedRoutes = cachedRoutes;
    }

    @Override
    public void intercept(Chain chain) throws RouteNotFoundException {
        String url = chain.url();
        if (cachedRoutes.get(url) != null) {
           // return cachedRoutes.get(url);
        }

        chain.proceed();
      //  cachedRoutes.put(url, routerParams);

    }

}

File: RouterLib/src/main/java/com/kronos/router/interceptor/CacheInterceptor.kt
package com.kronos.router.interceptor

import com.kronos.router.exception.RouteNotFoundException
import com.kronos.router.model.RouterParams

class CacheInterceptor internal constructor(private val cachedRoutes: HashMap<String, RouterParams>) : BaseInterceptor() {


    @Throws(RouteNotFoundException::class)
    override fun intercept(chain: Interceptor.Chain) {
        val url = chain.url.url
        if (cachedRoutes[url] != null) {
            val params = cachedRoutes[url]
            params?.apply {
                cachedRoutes[chain.url.url] = params
                pathInterceptor(this, chain)
            }
            return
        } else {
            chain.proceed()
        }
    }

}